packages/shell-api/src/shard.ts

```diff
@@ -464,4 +464,73 @@ export default class Shard extends ShellApiWithMongoClass {
 
     return cursor;
   }
+
+  @returnsPromise
+  @apiVersions([])
+  @serverVersions(['7.0.0', ServerVersions.latest])
+  async startAutoMerger(): Promise<UpdateResult> {
+    this._emitShardApiCall('startAutoMerger', {});
+    const config = await getConfigDB(this._database);
+    return await config.getCollection('settings').updateOne(
+      { _id: 'automerge' },
+      { $set: { enabled: true } },
+      { upsert: true, writeConcern: { w: 'majority', wtimeout: 30000 } }
+    ) as UpdateResult;
+  }
+
+  @returnsPromise
+  @apiVersions([])
+  @serverVersions(['7.0.0', ServerVersions.latest])
+  async stopAutoMerger(): Promise<UpdateResult> {
+    this._emitShardApiCall('stopAutoMerger', {});
+    const config = await getConfigDB(this._database);
+    return await config.getCollection('settings').updateOne(
+      { _id: 'automerge' },
+      { $set: { enabled: false } },
+      { upsert: true, writeConcern: { w: 'majority', wtimeout: 30000 } }
+    ) as UpdateResult;
+  }
+
+  @returnsPromise
+  @apiVersions([])
+  @serverVersions(['7.0.0', ServerVersions.latest])
+  async shouldAutoMerge(): Promise<boolean> {
```

Comment:
Did the sharding team come up with this name? I’m wondering if something like “sh.isAutoMergerEnabled()” is a better name, also considering that “sh.isBalancerRunning()” is a thing

Reply:
Yes I copied from where the ticket said I should copy from: https://github.com/10gen/mongo/pull/11291/files

Reply:
Yeah, that’s the nice thing about the legacy shell not shipping anymore … we can do whatever we think is best in terms of UX.